### DISCUSSION TITLE
Is it possible to use Markdoc with SolidJS?

### DISCUSSION THREAD

**Discussion by didimedina** [markdoc/markdoc#269] [Q&A] [answered]
Since SolidJS doesn't have a virtual dom and just interacts with dom nodes directly I'm wondering if Markdoc would work out the box with the standard [HTML renderer](https://markdoc.dev/docs/render#html). Does anyone try this already / have an example up somewhere I could reference? Thanks in advance 🙌🏼

**Comment by rpaul-stripe** [comment, marked as answer]
SolidJS has its own JSX compiler that generates a combination of code and HTML template strings. You need the generated code in order to use external components inside of the document, so you can't just use HTML string output. You need a custom renderer that can use SolidJS's [Dynamic](https://www.solidjs.com/docs/latest/api#dynamic) feature.

There's a third-party Markdoc renderer for SolidJS here: https://github.com/dillonchanis/solidjs-markdoc
